extension-manifest-v3/src/store/options.js

```diff
@@ -148,15 +148,15 @@ async function migrateFromMV2() {
   }
 }
 
-export function observe(property, fn) {
+export async function observe(property, fn) {
   let value;
-  const wrapper = (options) => {
+  const wrapper = async (options) => {
     if (value === undefined || options[property] !== value) {
       const prevValue = value;
       value = options[property];
 
       try {
-        fn(value, prevValue);
+        return fn(value, prevValue);
```

Comment:
return await fn(value, prevValue);
  
  otherwise, catch will not have an effect